tket/src/Architecture/Architecture.cpp

```diff
@@ -18,37 +18,26 @@
 #include <unordered_set>
 #include <vector>
 
-#include "Circuit/Conditional.hpp"
 #include "Graphs/ArticulationPoints.hpp"
 #include "Utils/Json.hpp"
 #include "Utils/UnitID.hpp"
 
 namespace tket {
 
-// basic implementation that works off same prior assumptions
-// TODO: Update this for more mature systems of multi-qubit gates
-bool Architecture::valid_operation(
-    const Op_ptr& op, const std::vector<Node>& uids) const {
-  if (op->get_desc().is_box() ||
-      (op->get_type() == OpType::Conditional &&
-       static_cast<const Conditional&>(*op).get_op()->get_desc().is_box())) {
-    return false;
+bool Architecture::valid_operation(const std::vector<Node>& uids) const {
+  for (Node n : uids) {
+    if (!this->node_exists(Node(n))) return false;
   }
   if (uids.size() == 1) {
-    // with current Architecture can assume all single qubit gates valid
-    return true;
-  } else if (op->get_type() == OpType::Barrier) {
     return true;
   } else if (uids.size() == 2) {
-    if (this->node_exists(uids[0]) && this->node_exists(uids[1]) &&
-        this->bidirectional_edge_exists(uids[0], uids[1])) {
+    if (this->bidirectional_edge_exists(uids[0], uids[1])) {
       return true;
     }
-  } else if (uids.size() == 3 && op->get_type() == OpType::BRIDGE) {
+  } else if (uids.size() == 3) {
```

Comment:
similalry, architectures in general dont support 3 qubit gates, so we should probably just return false if there are more than 2 uids

Reply:
will remove that